src/V3EmitCImp.cpp

```diff
@@ -19,10 +19,138 @@
 
 #include "V3Global.h"
 #include "V3EmitC.h"
+#include "V3Ast.h"
 #include "V3EmitCFunc.h"
+#include "V3String.h"
+#include "V3UniqueNameMap.h"
 
+#include <map>
+#include <set>
 #include <vector>
-#include <unordered_set>
+
+//######################################################################
+// Utility functions
+
+static const AstNodeModule* getModp(const AstNode* nodep) {
+    const AstNodeModule* const modp = VN_CAST_CONST(nodep->user4p(), NodeModule);
+    UASSERT_OBJ(modp, nodep, "Missing module pointer " << nodep);
+    return modp;
+}
+
+//######################################################################
+// Visitor that gathers the headers required by an AstCFunc
+
+class EmitCGatherDependencies final : AstNVisitor {
+    // Ordered set, as it is used as a key in another map.
+    std::set<string> m_dependencies;
+
+    // METHODS
+    void addSymsDependency() { m_dependencies.insert(EmitCBaseVisitor::symClassName()); }
+    void addModDependency(const AstNodeModule* modp) {
+        if (const AstClass* const classp = VN_CAST_CONST(modp, Class)) {
```

Comment:
If you prefer you can use "auto" in cases like these, as the cast makes it obvious what the type is.

Reply:
I kind of like to be able to read the declaration as C and know the type, without having to look at the initializer, so I usually refrain from using auto for trivial cases like this.